[FLINK-6164] Make ProcessWindowFunction a RichFunction

zentol opened the pull request:
This PR modifies the ProcessWindowFunction and ProcessAllWindowFunction classes to extend AbstractRichtFunction.

The existing Rich* versions were deprecated, and all usages of them replaced.

aljoscha commented on flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/function/ProcessAllWindowFunction.scala:
```diff
@@ -35,7 +35,9 @@ import org.apache.flink.util.Collector
   */
 @PublicEvolving
-abstract class ProcessAllWindowFunction[IN, OUT, W <: Window] extends Function with Serializable {
+abstract class ProcessAllWindowFunction[IN, OUT, W <: Window]
```
I think the `Serializable` is not needed here. (Was there before, so not introduced in this PR)

aljoscha commented on flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/function/ProcessWindowFunction.scala:
```diff
@@ -36,7 +36,10 @@ import org.apache.flink.util.Collector
   */
 @PublicEvolving
-abstract class ProcessWindowFunction[IN, OUT, KEY, W <: Window] extends Function with Serializable {
+abstract class ProcessWindowFunction[IN, OUT, KEY, W <: Window]
```
See above.

aljoscha commented on flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/AccumulatingAlignedProcessingTimeWindowOperatorTest.java:
```diff
@@ -1038,7 +1038,7 @@ private void assertInvalidParameter(long windowSize, long windowSlide) {
 	// ------------------------------------------------------------------------
 
-	private static class StatefulFunction extends RichProcessWindowFunction<Integer, Integer, Integer, TimeWindow> {
+	private static class StatefulFunction extends ProcessWindowFunction<Integer, Integer, Integer, TimeWindow> {
```
With this we add two unused imports.

And yes, I don't like that we don't have checkstyle active on the tests.

zentol commented:
I've removed Serializable, adjusted the `@Deprecated` annotations like in #3816 and fixed the unused imports (there were also some in `ProcessAllWindowFunction.scala`).

aljoscha commented:
Ha! I didn't think of the annotations.

This looks good to merge now!

Thanks for fixing these so quickly! 😃

zentol commented:
merging.
